### PR TITLE
Various Refactorings

### PR DESCRIPTION
**Pull request by jplatte**
Most of these are based on clippy suggestions.

### CODE REVIEW

**Comment by jplatte** [comment]
Ping @tekjar

**Comment by tekjar** [comment]
@jplatte Thank you. This repo is going through design changes internally. I'll incorporate these changes in the next major release

**Comment by jplatte** [comment]
Ping @tekjar, are things still moving?

**Comment by de-sh** [comment]
Hey @jplatte, sorry for the delay from our side. We had been unable to focus on rumqtt for a while due to certain unforeseen circumstance. I have spoken with @tekjar and will be taking over the responsibility of reviewing and merging your PR from him.

**Comment by jplatte** [comment]
The only changes that are unrelated to clippy are the first (manual refactoring, just for better readability) and third (clearly) commits. The only ones that could have an impact on runtime performance are using `&str` instead of `&String` (removal of an unnecessary pointer indirection) and removing unnecessary clones (removes unnecessary work, though likely just for the optimizer rather than the resulting program).

Compile time should not be affected in a meaningful way; readability / maintainability should be equivalent or improved for all of the changes.

File: rumqttlog/src/logs/acks.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ impl Acks {
         match qos {
             1 => self.acks.push(Packet::PubAck(PubAck::new(pkid))),
             2 => self.acks.push(Packet::PubRec(PubRec::new(pkid))),
-            _ => return,
+            _ => {}
```
Comment (de-sh) [inline rumqttlog/src/logs/acks.rs:41]: We might have missed a oxford `,` here?

File: benchmarks/clients/rumqttsync.rs [hunk 2]
```diff
@@ -35,8 +35,8 @@ pub fn start(id: &str, payload_size: usize, count: usize) -> Result<(), Box<dyn
 
     let mut acks_count = 0;
     let start = Instant::now();
-    for event in connection.iter() {
-        if let Ok(Event::Incoming(Incoming::PubAck(_))) = event {
+    for event in connection.iter().flatten() {
```
Comment (de-sh) [inline benchmarks/clients/rumqttsync.rs:38]: I had not known about `flatten()`, nice usage. Should we add a comment atop or will this be self-explanatory?

**Review by de-sh — CHANGES_REQUESTED** [review]
@jplatte I have added a few comments, but otherwise this looks ready to merge!

File: rumqttlog/src/logs/acks.rs (hunk 1 above)
Comment (jplatte) [reply rumqttlog/src/logs/acks.rs:41]: Commas after `{}` in match arms qre removed by rustfmt by default.

File: benchmarks/clients/rumqttsync.rs (hunk 2 above)
Comment (jplatte) [reply benchmarks/clients/rumqttsync.rs:38]: I can add a comment.

Comment (jplatte) [reply benchmarks/clients/rumqttsync.rs:38]: Actually I'll just revert this change, after thinking more about it I don't like it anymore.

**Comment by jplatte** [comment]
Looks like something broke that's unrelated to this PR. I'll try rebasing to see whether that helps.

**Review by de-sh — APPROVED** [review]
LGTM

**Comment by de-sh** [comment]
Thanks @jplatte 🍾

**Pull request merged**